frontend/widgets/OBSBasic_Transitions.cpp

```diff
@@ -1359,5 +1411,48 @@ int OBSBasic::GetOverrideTransitionDuration(OBSSource source)
 
 int OBSBasic::GetTransitionDuration()
 {
-	return ui->transitionDuration->value();
+	return transitionDuration;
+}
+
+void OBSBasic::SetCurrentTransition(const std::string &uuid, bool setTransition)
```

Comment:
I find function signatures like this unfortunate, if you read this in normal language it originally allowed the caller to say:

> "Set the current transition to the transition identified by this UUID"

The added boolean makes it seem as if the instruction now becomes

> "Set the current transition to the transition identified by this UUID and actually _trigger_ it"

Would that be a correct reading?

Reply:
Is the new name `UpdateCurrentTransition` okay ?

Reply:
`SetTransition` exist because the frontend API can accept a transition that is not part of OBS Studio own collection, otherwise some things would be a little simpler.

I thought about just duplicate `SetCurrentTransition` UUID side in `SetTransition` current transition update side to avoid the new parameter but I was split between code duplication or reuse.